compair/api/answer_comment.py

```diff
@@ -312,6 +312,13 @@ def post(self, course_uuid, assignment_uuid, answer_uuid, answer_comment_uuid):
         if comment_type not in comment_types:
             abort(400, title="Feedback Not Saved", message="This feedback type is not recognized. Please contact support for assistance.")
 
+        # do not allow changing a self-eval into a comment or vise-versa
+        if answer_comment.comment_type == AnswerCommentType.self_evaluation and AnswerCommentType(comment_type) != AnswerCommentType.self_evaluation:
+            abort(400, title="Feedback Not Saved", message="Feedback type cannot be changed. Please contact support for assistance.")
+
+        if answer_comment.comment_type != AnswerCommentType.self_evaluation and AnswerCommentType(comment_type) == AnswerCommentType.self_evaluation:
+            abort(400, title="Feedback Not Saved", message="Feedback type cannot be changed. Please contact support for assistance.")
+
```

Comment:
Please also add these checks for `answer_comment.comment_type == AnswerCommentType.evaluation` as well